static/src/javascripts/projects/common/modules/identity/email-preferences.js

```diff
@@ -0,0 +1,175 @@
+// @flow
+// This is a workaround for the email preferences page https://profile.thegulocal.com/email-prefs
+// We want to submit subscribe/unsubscribe requests without a full page refresh
+// Hopefully this will be short-lived; if it is still alive in 2017, git blame and cry
+
+import bean from 'bean';
+import reqwest from 'reqwest';
+import fastdom from 'fastdom';
+import $ from 'lib/$';
+
+const addUpdatingState = buttonEl => {
+    fastdom.write(() => {
+        buttonEl.disabled = true;
+        $(buttonEl).addClass('is-updating is-updating-subscriptions');
+    });
+};
+
+const generateFormQueryString = buttons => {
+    const csrfToken = $('.form')[0].elements.csrfToken.value.toString();
+    const htmlPreference = $('[name="htmlPreference"]:checked').val();
+    let buttonString = '';
+    for (let i = 0; i < buttons.length; i += 1) {
+        const value = buttons[i].value;
+        const unsubscribeMatches = value.match(/unsubscribe-(.*)/);
+        if (unsubscribeMatches) {
+            const listIds = unsubscribeMatches[1].split(',');
+            for (let j = 0; j < listIds.length; j += 1) {
+                buttonString += `removeEmailSubscriptions[]=${encodeURIComponent(
+                    listIds[j]
+                )}&`;
+            }
+        } else {
+            buttonString += `addEmailSubscriptions[]=${encodeURIComponent(
+                value
+            )}&`;
+        }
+    }
+    return `csrfToken=${encodeURIComponent(
+        csrfToken
+    )}&${buttonString}htmlPreference=${encodeURIComponent(htmlPreference)}`;
```

Comment:
Would it make sense to store the values of `encodeURIComponent()` before, to make this more readable?